src/wallet/migrate.cpp

```diff
@@ -0,0 +1,738 @@
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <compat/byteswap.h>
+#include <crypto/common.h> // For ReadBE32
+#include <logging.h>
+#include <streams.h>
+#include <uint256.h>
+#include <wallet/migrate.h>
+
+#include <optional>
+#include <variant>
+
+namespace wallet {
+// Magic bytes in both endianness's
+constexpr uint32_t BTREE_MAGIC = 0x00053162; // If the file endianness matches our system, we see this magic
+constexpr uint32_t BTREE_MAGIC_OE = 0x62310500; // If the file endianness is the other one, we will see this magic
+
+// Subdatabase name
+static const std::vector<std::byte> SUBDATABASE_NAME = {std::byte{'m'}, std::byte{'a'}, std::byte{'i'}, std::byte{'n'}};
+
+enum class PageType : uint8_t
+{
+    INVALID = 0,         // Invalid page type
+    DUPLICATE = 1,       // Duplicate. Deprecated and no longer used
+    HASH_UNSORTED = 2,   // Hash pages. Deprecated.
+    BTREE_INTERNAL = 3,  // BTree internal
+    RECNO_INTERNAL = 4,  // Recno internal
+    BTREE_LEAF = 5,      // BTree leaf
+    RECNO_LEAF = 6,      // Recno leaf
+    OVERFLOW_DATA = 7,        // Overflow
+    HASH_META = 8,       // Hash metadata
+    BTREE_META = 9,      // BTree metadata
+    QUEUE_META = 10,     // Queue Metadata
+    QUEUE_DATA = 11,     // Queue Data
+    DUPLICATE_LEAF = 12, // Off-page duplicate leaf
+    HASH_SORTED = 13,    // Sorted hash page
+
+    PAGETYPE_MAX = 14,
+};
+
+enum class RecordType : uint8_t
+{
+    KEYDATA = 1,
+    DUPLICATE = 2,
+    OVERFLOW_DATA = 3,
+    DELETE = 0x80, // Indicate this record is deleted. This is AND'd with the real type.
+};
+
+enum class BTreeFlags : uint32_t
+{
+    DUP = 1,         // Duplicates
+    RECNO = 2,       // Recno tree
+    RECNUM = 4,      // BTree: Maintain record counts
+    FIXEDLEN = 8,    // Recno: fixed length records
+    RENUMBER = 0x10, // Recno: renumber on insert/delete
+    SUBDB = 0x20,    // Subdatabases
+    DUPSORT = 0x40,  // Duplicates are sorted
+    COMPRESS = 0x80, // Compressed
+};
+
+/** Berkeley DB BTree metadata page layout */
+class MetaPage
+{
+public:
+    uint64_t lsn;             // Log Sequence Number
+    uint32_t page_num;        // Current page number
+    uint32_t magic;           // Magic number
+    uint32_t version;         // Version
+    uint32_t pagesize;        // Page size
+    uint8_t encrypt_algo;     // Encryption algorithm
+    PageType type;            // Page type
+    uint8_t metaflags;        // Meta-only flags
+    uint8_t unused1;          // Unused
+    uint32_t free_list;       // Free list page number
+    uint32_t last_page;       // Page number of last page in db
+    uint32_t partitions;      // Number of partitions
+    uint32_t key_count;       // Cached key count
+    uint32_t record_count;    // Cached record count
+    BTreeFlags flags;         // Flags
+    uint160 uid;              // Unique file ID (20 bytes, fits in uint16)
```

Comment:
i don't think importing `uint160` is necessary here--the actual value isnt used, we're just using it to consume some bytes

Reply:
Dropped `uint160`, changed to use an array.